File: FRONT/AmusementPark/src/app/core/seo/seo-page-value-policy.ts
export const MINIMUM_INDEXABLE_IMAGE_GALLERY_ENTRIES: number = 3;
export const MINIMUM_INDEXABLE_COLLECTION_ENTRIES: number = 2;

export function isImageGalleryIndexable(entryCount: number): boolean {
  return entryCount >= MINIMUM_INDEXABLE_IMAGE_GALLERY_ENTRIES;
}

export function isCollectionIndexable(entryCount: number): boolean {
  return entryCount >= MINIMUM_INDEXABLE_COLLECTION_ENTRIES;
}

File: API/AmusementPark.Application/Features/Seo/Services/SeoPageValuePolicy.cs
namespace AmusementPark.Application.Features.Seo.Services;

/// <summary>
/// Centralise les seuils minimaux des pages publiques de collection qui peuvent etre indexees.
/// Les memes seuils sont reproduits dans la politique frontend afin que les balises robots et
/// les sitemaps restent alignes.
/// </summary>
public static class SeoPageValuePolicy
{
    public const int MinimumImageGalleryEntries = 3;

    public const int MinimumCollectionEntries = 2;

    public static bool IsImageGalleryIndexable(int entryCount)
    {
        return entryCount >= MinimumImageGalleryEntries;
    }

    public static bool IsCollectionIndexable(int entryCount)
    {
        return entryCount >= MinimumCollectionEntries;
    }
}
